### PR TITLE
Remove go1.6/go1.7 support

### PR DESCRIPTION
**Pull request by thinkerou**
closing #1381

### CODE REVIEW

File: debug_test.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func TestDebugPrintWARNINGDefault(t *testing.T) {
 	})
 	m, e := getMinVer(runtime.Version())
 	if e == nil && m <= ginSupportMinGoVer {
-		assert.Equal(t, "[GIN-debug] [WARNING] Now Gin requires Go 1.6 or later and Go 1.7 will be required soon.\n\n[GIN-debug] [WARNING] Creating an Engine instance with the Logger and Recovery middleware already attached.\n\n", re)
+		assert.Equal(t, "[GIN-debug] [WARNING] Now Gin requires Go 1.7 or later and Go 1.8 will be required soon.\n\n[GIN-debug] [WARNING] Creating an Engine instance with the Logger and Recovery middleware already attached.\n\n", re)
```
Comment (appleboy) [inline debug_test.go]: Drop go 1.7 version?

Comment (thinkerou) [reply debug_test.go]: yes, thanks!

**Comment by thinkerou** [comment]
The COV drop was caused by some code only used on go1.6/1.7, maybe we should find and remove it.

File: context.go [hunk 2]
```diff
@@ -419,11 +419,6 @@ func (c *Context) GetPostFormArray(key string) ([]string, bool) {
 	if values := req.PostForm[key]; len(values) > 0 {
 		return values, true
 	}
-	if req.MultipartForm != nil && req.MultipartForm.File != nil {
```
Comment (thinkerou) [inline context.go]: the branch used only go1.6

**Comment by thinkerou** [comment]
@javierprovecho @appleboy we need to remove go1.6/1.7 support at gin1.4 version? thanks!

**Comment by thinkerou** [comment]
I add unit test for `http2 pusher`, but seem invalid, I don't know why.

**Comment by thinkerou** [comment]
Hi @appleboy the pull request need to merge to gin1.4 release version? thanks!

**Comment by thinkerou** [comment]
see #1851

**Pull request closed without merging**